src/commands/delete.ts

```diff
@@ -53,26 +53,26 @@ function getItemToDelete(contextItem: TektonNode, selectedItems: TektonNode[]):
 
 async function doDelete(items: TektonNode[], toRefresh: Refreshable, commandId?: string): Promise<void | string> {
   if (items) {
-    const taskRunList = await getTaskRunResourceList();
     let message: string;
-    let refResource = true;
+    let hasAnyReference = true;
+    const pipelineList = await getPipelineList();
     const toDelete = new Map<TektonNode, CliCommand>();
     for (const item of items) {
       const deleteCommand = getDeleteCommand(item);
       if (deleteCommand) {
         toDelete.set(item, deleteCommand);
       }
-      if (refResource && checkRefResource()) {
-        if (referenceOfTaskAndClusterTaskInCluster(item, taskRunList)) {
-          refResource = false;
+      if (hasAnyReference && checkRefResource() && pipelineList.length !== 0) {
+        if (referenceOfTaskAndClusterTaskInCluster(item, pipelineList) && (item.contextValue === ContextType.TASK || item.contextValue === ContextType.CLUSTERTASK)) {
```

Comment:
The check if the item is a task or ct should be made before calling the `referenceOfTaskAndClusterTaskInCluster`

Something like
```
 if (!hasAnyReference && pipelineList.length !== 0 && (item.contextValue === ContextType.TASK || item.contextValue === ContextType.CLUSTERTASK)) {
  hasAnyReference = referenceOfTaskAndClusterTaskInCluster(item, pipelineList)
```